### PR TITLE
Added proper callback handling

### PR DESCRIPTION
**Pull request by mkotsollaris**

### CODE REVIEW

File: src/TruliooClient.js [hunk 1]
```diff
@@ -1,30 +1,73 @@
 export default class TruliooClient {
 
     constructor(obj) {
+
         for (var fld in obj) {
             this[fld] = obj[fld];
         }
+        this.embedIDURL = !this.embedIDURL ? "http://localhost:8855/embedid" : this.embedIDURL;
```
Comment (kateika) [inline src/TruliooClient.js]: please revert logic: 
this.embedIDURL = this.embedIDURL ?  this.embedIDURL : "http://localhost:8855/embedid";

File: src/TruliooClient.js [hunk 2]
```diff
@@ -1,30 +1,73 @@
...
+            console.log('something went wrong during EmbedID form initialization', error);
+        }
+    }
+
+    /**
+    * Registers handleSubmit and handleResponse callbacks.
+    */
+    registerEvents() {
+        const truliooClient = this;
+        if (window && window.addEventListener) {
```
Comment (kateika) [inline src/TruliooClient.js]: don't see removing of event listener. Do we need to remove listener?

File: src/TruliooClient.js [hunk 3]
```diff
@@ -1,30 +1,73 @@
...
+            window.addEventListener("message", onMessage, false);
+        }
+
+        function onMessage(event) {
+            try {
+                const eventOriginCompleteURL = `${event && event.origin}/embedid`;
```
Comment (kateika) [inline src/TruliooClient.js]: what type of event.origin? string/object? Could it be undefined or null by accident?

File: src/TruliooClient.js [hunk 4]
```diff
@@ -1,30 +1,73 @@
...
-        const response = await fetch(originURL, {
-            method: 'POST'
-        });
-        const deconstructedResult = await response.json();
-        const accessToken = deconstructedResult.accessToken;
-        this['accessToken'] = accessToken;
+        try {
+            const originURL = `${this.accessTokenURL}/${this.publicKey}`;
+            const response = await fetch(originURL, {
+                method: 'POST'
```
Comment (kateika) [inline src/TruliooClient.js]: we don't need any body for this post?

File: example.html [hunk 5]
```diff
@@ -7,11 +7,19 @@
...
-        const publicKey = 'PUBLIC_KEY_GOES_HERE'; // Public Key
-        const accessTokenURL = 'ACCESS_TOKEN_URL_GOES_HERE'; // Access Token URL
+        function handleResponse(e) {
+            console.log('@handleResponse', e);
+        }
+        function handleSubmit(e) {
+            console.log('@handleSubmit', e);
+        }
+        const publicKey = 'PUT_YOUR_PUBLIC_KEY_HERE'; // Public Key
+        const accessTokenURL = ' PUT_YOUR_ACCESS_TOKEN_URL_HERE'; // Access Token URL
```
Comment (kateika) [inline example.html]: extra space before PUT should be deleted

Comment (kateika) [reply example.html]: I thought that at some point we decided not to have url with publicKey and accessTokenURL? I thought we were thinking just about accessToken without a key? Not sure though

File: src/TruliooClient.js (hunk 4 above)
Comment (mkotsollaris) [reply src/TruliooClient.js]: that's right, we don't

File: src/TruliooClient.js (hunk 1 above)
Comment (mkotsollaris) [reply src/TruliooClient.js]: Thanks! ✅

File: src/TruliooClient.js (hunk 2 above)
Comment (mkotsollaris) [reply src/TruliooClient.js]: unless we are adding a custom lifecycle event i.e.: `destroyEmbedID`, there's no really a clear way to know when to remove this listener.
Most of the up to date browsers will remove this by default, so I think we should be safe around potential memory leaks.

File: src/TruliooClient.js (hunk 3 above)
Comment (mkotsollaris) [reply src/TruliooClient.js]: I doubt event.origin will ever be `null/undefined`.

File: src/TruliooClient.js [hunk 6]
```diff
@@ -1,30 +1,73 @@
...
 
     async init() {
-        await this.injectAccessToken();
-        this.loadEmbedID();
+        try {
+            await this.injectAccessToken();
+            this.registerEvents();
+            this.loadEmbedID();
+        } catch (error) {
+            console.log('something went wrong during EmbedID form initialization', error);
```
Comment (VictoriaGreen93) [inline src/TruliooClient.js]: we should at least error an error rather than log it

File: src/TruliooClient.js [hunk 7]
```diff
@@ -1,30 +1,73 @@
...
+                var data = event.data;
+
+                // call function
+                if (truliooClient[data.function]) {
+                    truliooClient[data.function].call(window, JSON.parse(data.message));
+                } else {
+                    console.log(`Tried to trigger ${data.func} but found no available callBack`);
+                }
+            } catch (error) {
+                console.log('Something went wrong during callback registration', error);
```
Comment (VictoriaGreen93) [inline src/TruliooClient.js]: again

File: src/TruliooClient.js [hunk 8]
```diff
@@ -1,30 +1,73 @@
...
+        try {
+            const originURL = `${this.accessTokenURL}/${this.publicKey}`;
+            const response = await fetch(originURL, {
+                method: 'POST'
+            });
+            const deconstructedResult = await response.json();
+            const accessToken = deconstructedResult.accessToken;
+            this.accessToken = accessToken;
+        } catch (err) {
+            console.log('Something went wrong during access token generation', err);
```
Comment (VictoriaGreen93) [inline src/TruliooClient.js]: and here

Comment (mkotsollaris) [reply src/TruliooClient.js]: 👍

**Pull request merged**